### PR TITLE
Hyperv : Fix issues in Storage test cases

### PR DESCRIPTION
**Pull request by asofron**
Issues addressed : 
**- Fixed sdcount to give correct output :** 
Sdcount now return correct devices count when compared to the ones in contants.sh 
**- Fixed failing vhdx cleanup :** 
Added cleanup script for vhd at the end of the testcases 
**- Fixed Hot Add vhd/vhdx :**
Test cases should use an existing controller and only hot add drives 
Affected Testcases : 

- VHDX-RESIZE-TO-BIGGER512
- VHDX-RESIZE-TO-BIGGER512-FIXED-IDE
- VHDX-RESIZE-GROWSHRINK-SCSI
- VHDX-RESIZE-GROWOVER2TB-SHRINK-SCSI-OFFLINE
- VHDX-RESIZE-GROWOVER2TB-SHRINK-IDE
- STORAGE-VHD-HOT-ADD-MULTI-DYNAMIC-SCSI
- STORAGE-VHDX-HOT-ADD-MULTI-DYNAMIC-SCSI
- STORAGE-VHDX-4K-HOT-ADD-MULTI-DYNAMIC-SCSI

```Test Run On           : 02/21/2019 18:02:57
VHD Under Test        : Ubuntu14.04.5-server.vhdx
Total Test Cases      : 8 (8 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:53

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 VHDX-RESIZE-TO-BIGGER512                                                          PASS                 3.66 
    2 VHDX-RESIZE-TO-BIGGER512-FIXED-IDE                                                PASS                 6.19 
    3 VHDX-RESIZE-GROWSHRINK-SCSI                                                       PASS                 4.83 
    4 VHDX-RESIZE-GROWOVER2TB-SHRINK-SCSI-OFFLINE                                       PASS                12.07 
    5 VHDX-RESIZE-GROWOVER2TB-SHRINK-IDE                                                PASS                11.92 
    6 STORAGE-VHD-HOT-ADD-MULTI-DYNAMIC-SCSI                                            PASS                 2.53 
    7 STORAGE-VHDX-HOT-ADD-MULTI-DYNAMIC-SCSI                                           PASS                 3.88 
    8 STORAGE-VHDX-4K-HOT-ADD-MULTI-DYNAMIC-SCSI                                        PASS                 3.88 
```

### CODE REVIEW

**Comment by asofron** [comment]
Need to check the all others test cases theat use STOR-Lis-Disk.sh as well

File: Testscripts/Windows/STOR-VHDX-RESIZE-GROWSHRINK.ps1 [hunk 1]
```diff
@@ -180,7 +180,6 @@ Function Main
 	} finally {
 		Stop-VM -VMName $vmName -ComputerName $hvServer -force
 		Remove-VMHardDiskDrive -VMHardDiskDrive $vhdxDrive
-		Remove-Item $vhdPath
```
Comment (juhlee-microsoft) [inline Testscripts/Windows/STOR-VHDX-RESIZE-GROWSHRINK.ps1]: Can you explain to me why it is removed in finally block?

File: Testscripts/Linux/STOR-Lis-Disk.sh [hunk 2]
```diff
@@ -59,8 +59,7 @@ Integrity_Check() {
 SetTestStateRunning
 # Count the number of SCSI= and IDE= entries in constants
 #
-disk_count=1
-# Set to 1 because of resource disk
```
Comment (lizzha) [inline Testscripts/Linux/STOR-Lis-Disk.sh]: is resource disk changed?

Comment (asofron) [reply Testscripts/Linux/STOR-Lis-Disk.sh]: disk_count gets the disks from constants.sh ( more exact the test params) , so if 1 the disk_count to sd_count will never be equal

File: Testscripts/Windows/STOR-VHDX-RESIZE-GROWSHRINK.ps1 (hunk 1 above)
Comment (asofron) [reply Testscripts/Windows/STOR-VHDX-RESIZE-GROWSHRINK.ps1]: Using cleanup script at the end to remove

**Pull request merged**